docs/source/data/data_catalog_yaml_examples.md

```diff
@@ -398,6 +402,78 @@ In this example, Kedro understands that `my_dataframe` is the same dataset in it
 In the pipeline, Kedro uses the `spark.SparkDataset` implementation for saving and `pandas.ParquetDataset`
 for loading, so the first node outputs a `pyspark.sql.DataFrame`, while the second node receives a `pandas.Dataframe`.
 
+### How *not* to use transcoding
+
+Kedro pipelines automatically resolve the node execution order and check to ensure there are no circular dependencies in the pipeline. It is during this process that the transcoded datasets are resolved and the transcoding notation `@...` is stripped. This means within the pipeline the datasets `my_dataframe@spark` and `my_dataframe@pandas` are considered to be one `my_dataframe` dataset. The `DataCatalog`, however, treats transcoded entries as separate datasets, as they are only resolved as part of the pipeline resolution process. This results in differences between your defined pipeline in `pipeline.py` and the resolved pipeline that is run by Kedro, and these differences may lead to unintended behaviours. Thus, it is important to be aware of this when using transcoding.
+
+Below are some examples where transcoding may produce unwanted side effects.
+
+#### Defining a node with the same inputs and outputs
```

Comment:
Is this still possible after the change?

Reply:
It's not possible, and this section explains why. Below, there's also an example of an error raising.

Reply:
I would put and admonition like this to make it clear to someone skim reading this quickly

> [!CAUTION]
> Put some content here .....